### ISSUE TITLE
2000 VM limit with getallvms.py

### ISSUE THREAD

**Issue by andrewfraley** [vmware/pyvmomi-community-samples#237]
I'm not sure if this is an issue with pyvmomi, the sample code, or a limitation of the vsphere api, but when using the [getallvms.py](https://github.com/vmware/pyvmomi-community-samples/blob/master/samples/getallvms.py) example I can only get exactly 2000 VMs (we have a bit more than that), which sounds like a hard coded limit somewhere.

**Comment by michaelrice** [comment]
That is honestly a horrible example for an inventory over 20 or so vms. Its slow and kludgy. You should use [vminfo_quick](https://github.com/vmware/pyvmomi-community-samples/blob/master/samples/vminfo_quick.py) if you have a large inventory. My guess on that sample is its not diving into folders correctly.

**Comment by andrewfraley** [comment]
@michaelrice you're right, vminfo_quick is much faster, however, I'm still bumping up against that 2000 vm limit.

**Comment by andrewfraley** [comment]
Upon further investigation I'm getting slightly less than 2000 with PowerCLI, so there may actually just be exactly 2000 VMs in this environment.
